lib/mixins/properties-mixin.html

```diff
@@ -0,0 +1,208 @@
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../utils/boot.html">
+<link rel="import" href="../utils/mixin.html">
+<link rel="import" href="../mixins/properties-changed.html">
+
+<script>
+(function() {
+  'use strict';
+
+  /**
+   * Mixin that provides a minimal starting point to using the PropertiesChanged
+   * mixin by providing a mechanism to declare properties in a static
+   * getter (e.g. static get properties() { return { foo: String } }). Changes
+   * are reported via the `_propertiesChanged` method.
+   *
+   * This mixin provides no specific support for rendering. Users are expected
+   * to create a ShadowRoot and put content into it and update it in whatever
+   * way makes sense. This can be done in reaction to properties changing by
+   * implementing `_propertiesChanged`.
+   *
+   * @mixinFunction
+   * @polymer
+   * @appliesMixin Polymer.PropertiesChanged
+   * @memberof Polymer
+   * @summary Mixin that provides a minimal starting point for using
+   * the PropertiesChanged mixin by providing a declarative `properties` object.
+   */
+   Polymer.PropertiesMixin = Polymer.dedupingMixin(superClass => {
+
+    /**
+     * @constructor
+     * @extends {superClass}
+     * @implements {Polymer_PropertiesChanged}
+     */
+    const base = Polymer.PropertiesChanged(superClass);
+
+    /**
+     * Mixes `moreProps` into `props` but upgrades shorthand type
+     * syntax to { type: Type}.
+     *
+     * @param {Object} props Properties to normalize
+     * @return {Object} Copy of input `props` with normalized properties that
+     * are in the form {type: Type}
+     * @private
+     */
+    function normalizeProperties(props) {
```

Comment:
This can move out of the mixin function to avoid creating a new function object per mixin application.

Reply:
Yep, will do.